src/main.ts

```diff
@@ -1,114 +1,59 @@
 
-    // map for different platform and arch
-    const extensionMap = {
-      linux: '.tar.gz',
-      darwin: '.zip',
-      win32: '.zip'
+    let windowsInstallScript = `powershell -ex AllSigned -c "Invoke-RestMethod 'https://aka.ms/install-azd.ps1' | Invoke-Expression"`
+    let linuxOrMacOSInstallScript = `sudo curl -fsSL https://aka.ms/install-azd.sh | bash`
+    if (version !== 'latest') {
+      windowsInstallScript = `powershell -ex AllSigned -c "Invoke-RestMethod 'https://aka.ms/install-azd.ps1' -OutFile 'install-azd.ps1'; powershell -ExecutionPolicy Bypass -File ./install-azd.ps1 -Version '${version}'"`
+      linuxOrMacOSInstallScript = `sudo curl -fsSL https://aka.ms/install-azd.sh | bash -s -- --version ${version}`
     }
 
-    const exeMap = {
-      linux: '',
-      darwin: '',
-      win32: '.exe'
-    }
-
-    const arm64Map = {
-      x64: 'amd64',
-      arm64: 'arm64-beta'
-    }
-
-    const platformMap = {
-      linux: 'linux',
-      darwin: 'darwin',
-      win32: 'windows'
+    core.info(`Installing azd version ${version} on ${os}.\n`)
+
+    if (os === 'win32') {
+      cp.execSync(windowsInstallScript)
+      // Add azd to PATH
+      if (localAppDataPath) {
+        const azdPath = path.join(localAppDataPath, 'Programs', 'Azure Dev CLI');
+        fs.appendFileSync(process.env.GITHUB_PATH || '', `${azdPath}${path.delimiter}`);
+      } else {
+        core.setFailed('LocalAppData environment variable is not defined.');
+      }
+    } else {
+      cp.execSync(linuxOrMacOSInstallScript)
     }
 
-    // get install url
-    const installArray = installUrlForOS(
-      os,
-      architecture,
-      platformMap,
-      arm64Map,
-      extensionMap,
-      exeMap
-    )
-
-    const url = `https://azd-release-gfgac2cmf7b8cuay.b02.azurefd.net/azd/standalone/release/${version}/${installArray[0]}`
-
     core.notice(`The Azure Developer CLI collects usage data and sends that usage data to Microsoft in order to help us improve your experience.
 You can opt-out of telemetry by setting the AZURE_DEV_COLLECT_TELEMETRY environment variable to 'no' in the shell you use.
-
 Read more about Azure Developer CLI telemetry: https://github.com/Azure/azure-dev#data-collection`)
 
-    core.info(`Installing azd from ${url}`)
-
-    const file = await toolCache.downloadTool(url)
-    let extracted
-    if (os !== 'linux') {
-      extracted = await toolCache.extractZip(file)
-    } else {
-      extracted = await toolCache.extractTar(file)
-    }
-
-    if (os !== 'win32') {
-      fs.symlinkSync(
-        path.join(extracted, installArray[1]),
-        path.join(extracted, 'azd')
-      )
+    // Run `azd version` so we get the version that was installed written to the log.
+    if (os === 'win32') {
+      if (localAppDataPath) {
+        const azdExePath = path.join(localAppDataPath, 'Programs', 'Azure Dev CLI', 'azd.exe');
+        core.info(`\nChecking azd version: `+cp.execSync(`"${azdExePath}" version`).toString())
```

Comment:
should `cp.execSync(`"${azdExePath}" version`).toString())` be inside a try/catch ?  or should we check the return code to see that it ran OK?  
What happen if it fails?  does it throw or something? or is it a silent fail?